sdk/src/androidTest/java/com/bugsnag/android/ClientNotifyTest.java

```diff
@@ -72,7 +77,55 @@ public void beforeNotify(Report report) {
         assertEquals("Message", error.getExceptionMessage());
     }
 
+    @Test
+    public void testNotifyAsyncMetadata() throws Exception {
+        MetaData metaData = new MetaData();
+        metaData.addToTab("animals", "dog", true);
+
+        client.notify(new RuntimeException("Foo"), metaData);
+        apiClient.awaitReport();
```

Comment:
Are you testing purpose of the Async functionality i.e. The call to `client.notify` is non-blocking anywhere?

Reply:
That's a good point, I'm not aware of a direct test for that anywhere. I can add in an assertion to check that the report is null after calling notify, but before calling the countdown latch.